Documentation/learning/glossary.md

```diff
@@ -26,10 +74,36 @@ Peer is another member of the same cluster.
 
 A proposal is a request (for example a write request, a configuration change request) that needs to go through raft protocol.
 
-## Client
+## Quorum
 
-Client is a caller of the cluster's HTTP API.
+The number of members of a cluster required to be available to perform requests (except for local request when specified). Quorum is the majority (N/2 + 1) of the cluster. For example, when there are 5 members in the cluster, the quorum is 3. When there are 6 members, the quorum will be 4. When there are 7 members, the quorum is still 4.
+
+## Revision
+
+Revision is the current revision of etcd. Every time a key gets updated/deleted the modRevision gets updated to the current revision of the store.
+
+## Role
+
+Role is used to specify differnt roles which can be assigned to etcd user(s). The roles assigned determines the set of operations which a user is eligible to perform.
+
+## Snapshot
+
+Snapshot is a process of making a copy of the etcd backend database to a file. The snapshot file can then be re-used to migrate the data to any other independent cluster or can be used to bring an etcd cluster member to the saved point.
+
+## Store
+
+Store is a place in etcd server which stores key-value pairs.
+
+## Transaction
+
+Transaction is a set of operations that are to be executed atomically based on a certain set of conditions.
```

Comment:
I don't think the set of conditions stuff is necessary since people will see that when they try to create a transaction.

`An atomically executed set of operations. All modified keys in a transaction share the same modification revision.`